sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -349,6 +333,13 @@ public void done() {
         throw new RetriableException("Error flushing " + tp, e);
       }
 
+      offsetsToCommit.put(tp, new OffsetAndMetadata(lastRecord.kafkaOffset() + 1));
```

Comment:
For me it wasn't straightforward that we need +1 to start from the next (unprocessed) message after restart. I'd leave a comment.

Reply:
Added a comment.